Core.UnitTests/CodeGeneration/ReflectionEmit/ModuleBuilderFactoryTest.cs

```diff
@@ -96,23 +110,35 @@ public void CreateModuleBuilder_StrongNamed_ProvidedKey ()
           c_assemblyName, assemblyDirectoryOrNull: null, strongNamed: true, keyFilePathOrNull: otherKeyPath);
 
       var publicKey = new StrongNameKeyPair (File.ReadAllBytes (otherKeyPath)).PublicKey;
-      CheckAdapterAndSaveToDiskBehavior (result, _currentDirectory, expectedPublicKey: publicKey);
+
+      CheckAdapterBehavior (result, expectedPublicKey: publicKey);
+#if FEATURE_ASSEMBLYBUILDER_SAVE
+      CheckSaveToDiskBehavior (result, _currentDirectory);
+#endif
     }
 
-    private void CheckAdapterAndSaveToDiskBehavior (IModuleBuilder moduleBuilder, string assemblyDirectory, byte[] expectedPublicKey = null)
+    private void CheckAdapterBehavior (IModuleBuilder moduleBuilder, byte[] expectedPublicKey = null)
     {
       Assert.That (moduleBuilder, Is.TypeOf<ModuleBuilderAdapter>());
       var moduleBuilderAdapter = (ModuleBuilderAdapter) moduleBuilder;
       Assert.That (moduleBuilderAdapter.ScopeName, Is.EqualTo (c_assemblyFileName));
       Assert.That (moduleBuilderAdapter.AssemblyBuilder, Is.TypeOf<AssemblyBuilderAdapter>());
       var assemblyBuilderAdapter = (AssemblyBuilderAdapter) moduleBuilder.AssemblyBuilder;
+
       Assert.That (assemblyBuilderAdapter.AssemblyName, Is.EqualTo (c_assemblyName));
       Assert.That (assemblyBuilderAdapter.PublicKey, Is.EqualTo (expectedPublicKey ?? new byte[0]));
+    }
+
+    private void CheckSaveToDiskBehavior (IModuleBuilder moduleBuilder, string assemblyDirectory)
+    {
+      Assert.That (moduleBuilder, Is.TypeOf<ModuleBuilderAdapter>());
+      var moduleBuilderAdapter = (ModuleBuilderAdapter) moduleBuilder;
+      Assert.That (moduleBuilderAdapter.ScopeName, Is.EqualTo (c_assemblyFileName));
+      Assert.That (moduleBuilderAdapter.AssemblyBuilder, Is.TypeOf<AssemblyBuilderAdapter>());
+      var assemblyBuilderAdapter = (AssemblyBuilderAdapter) moduleBuilder.AssemblyBuilder;
 
       var assemblyPath = Path.Combine (assemblyDirectory, c_assemblyFileName);
       var pdbPath = Path.Combine (assemblyDirectory, c_pdbFileName);
-      Assert.That (File.Exists (assemblyPath), Is.False);
-      Assert.That (File.Exists (pdbPath), Is.False);
 
       var result = assemblyBuilderAdapter.SaveToDisk();
```

Comment:
```suggestion
      var result = moduleBuilder.AssemblyBuilder.SaveToDisk();
```

Given the new test-layout, we no longer need to do a cast to AssemblyBuilderAdapter but isntead we can just use SaveToDisk from the interface. I normally wouldn't do this change, but it allows us to remove the redundant assertions in line 137/138.